### DISCUSSION TITLE
TemporaryUploadedFile issue (Filepond)

### DISCUSSION THREAD

**Discussion by madurapa** [spatie/laravel-livewire-wizard#61] [General]
This issue occurs when going back and forth through the steps: https://github.com/livewire/livewire/discussions/3134

For additional details, kindly refer to this pull request as well: https://github.com/livewire/livewire/pull/2212

**Comment by ju5t** [comment]
The references you posted point at issues in the Livewire codebase, not this one. So I am not sure what your question is here.

**Reply by liamseys** [reply]
I have the same issue. The question here is how we can add adding images through the steps. I know it might not be part of the package but I would like to know any thoughts on the best practices to implement such a functionality.

**Comment by madurapa** [comment]
This is the form with which I am dealing with the problem (using Filepond but checking with the native file field as well the result is the same though)
![image](https://user-images.githubusercontent.com/4289578/205468475-00bf7a73-6651-4302-b758-dd37920a85d9.png)

In the files property, we now have a TemporaryUploadedFile instance.
![image](https://user-images.githubusercontent.com/4289578/205468651-36f03c9d-fca8-46ef-b2d4-8eb0509e4a24.png)

The uploaded files vanished when I clicked the Back button and then returned by clicking Next.
![image](https://user-images.githubusercontent.com/4289578/205468815-8049e819-76cd-47b2-966c-d605fc3e1129.png)

And the files property now looks like this:
![image](https://user-images.githubusercontent.com/4289578/205468855-6f4ef2a0-f3bd-41a3-a71a-88d2d8c2f38f.png)

**Reply by liamseys** [reply]
Yes I have the exact same issue. There is a hacky way to fix it but I am still looking for some advice on best practices.

**Reply by ju5t** [reply]
So, if I understand things correctly, the issue originates in Livewire as it's not serialising uploaded files in events. And as the Livewire Wizard is an event driven package, you can't upload files in any of the steps. In the future, please explain the issue instead of referencing links, it helps to get a better understanding quicker.

There are no best practices in this package as long as Livewire doesn't serialise `TemporaryUploadedFile`'s. It will always be a workaround. Personally, I would probably do something like this:

Upload file step:

```php
public function nextStep()
{
    // Serialise the files public property.
    parent::nextStep();
}
```

Final step (or any step where you need those files)

```php
public function mount()
{
    $this->files = $this->state()->forStep(Livewire::getAlias(UploadFileStep::class));
    // Transform back into a `TemporaryUploadedFile` object.
}
```
